packages/vm/tests/api/runBlock.js

```diff
@@ -146,6 +146,74 @@ tape('should run valid block', async (t) => {
+  const defaultCommon = new Common('mainnet', "dao")
+  // retrieve the hard forks list from defaultCommon...
+  let forks = defaultCommon.hardforks()
+  let editedForks = []
+  // explicitly edit the "dao" block number:
+  for (let fork of forks) {
+    if (fork.name == "dao") {
+      editedForks.push({
+        name: "dao",
+        forkHash: fork.forkHash,
+        block: 1
+      })
+    } else {
+      editedForks.push(fork)
+    }
+  }
+  let common = Common.forCustomChain('mainnet', {
+    hardforks: editedForks
+  }, 'homestead') // we should be on the "homestead" fork
+
+  const vm = setupVM({ common })
+  const suite = setup(vm)
+
+  const genesis = new Block(util.rlp.decode(suite.data.genesisRLP))
+  let block1 = util.rlp.decode(suite.data.blocks[0].rlp)
+  // edit extra data of this block to "dao-hard-fork"
+  block1[0][12] = Buffer.from('dao-hard-fork')
+  const block = new Block(block1)
+  await setupPreConditions(suite.vm.stateManager._trie, suite.data)
+
+  // fill two original DAO child-contracts with funds and the recovery account with funds in order to verify that the balance gets summed correctly
+  let fundBalance1 = Buffer.from('1111', 'hex') 
+  let accountFunded1 = createAccount(Buffer.from('', 'hex'), fundBalance1)
+  let DAOFundedContractAddress1 = Buffer.from("d4fe7bc31cedb7bfb8a345f31e668033056b2728", 'hex')
+  await suite.vm.stateManager.putAccount(DAOFundedContractAddress1, accountFunded1)
+
+  let fundBalance2 = Buffer.from('2222', 'hex') 
+  let accountFunded2 = createAccount(Buffer.from('', 'hex'), fundBalance2)
+  let DAOFundedContractAddress2 = Buffer.from("b3fb0e5aba0e20e5c49d252dfd30e102b171a425", 'hex')
+  await suite.vm.stateManager.putAccount(DAOFundedContractAddress2, accountFunded2)
+
+  let DAORefundAddress = Buffer.from("bf4ed7b27f1d666546e30d74d50d173d20bca754", 'hex')
+  let fundBalanceRefund = Buffer.from('4444', 'hex') 
+  let accountRefund = createAccount(Buffer.from('', 'hex'), fundBalanceRefund)
+  await suite.vm.stateManager.putAccount(DAORefundAddress, accountRefund)
+
+  let res = await suite.vm.runBlock({
+    block,
+    skipBlockValidation: true,
+    generate: true
+  })
+
+  t.error(res.error, "runBlock shouldn't have returned error")
+
+  let DAOFundedContractAccount1 = await suite.vm.stateManager.getAccount(DAOFundedContractAddress1)
+  t.true(DAOFundedContractAccount1.balance.equals(Buffer.from('', 'hex'))) // verify our funded account is now has 0 balance
+  let DAOFundedContractAccount2 = await suite.vm.stateManager.getAccount(DAOFundedContractAddress2)
+  t.true(DAOFundedContractAccount2.balance.equals(Buffer.from('', 'hex'))) // verify our funded account is now has 0 balance
+
+  let DAORefundAccount = await suite.vm.stateManager.getAccount(DAORefundAddress)
+  t.true(DAORefundAccount.balance.equals(Buffer.from('7777', 'hex'))) // verify that the refund account gets the summed balance of the original refund account + two child DAO accounts
+
+  t.end()
```

Comment:
Nice tests, really cool that you went this extra path! 😀  Small typo in both of the `// verify our funded account is now has 0 balance` comments (remove the `is`).